groth16/src/groth16.rs

```diff
@@ -0,0 +1,188 @@
+use crate::bellman_ce::{groth16::*, Circuit};
+use crate::snark::SNARK;
+use algebraic::errors::Result;
+use franklin_crypto::bellman::pairing::Engine;
+use rand::Rng;
+use std::marker::PhantomData;
+
+pub struct Groth16<E: Engine, C: Circuit<E>> {
+    _engine: PhantomData<E>,
+    _circuit: PhantomData<C>,
+}
+
+impl<E: Engine, C: Circuit<E>> SNARK<E::Fr> for Groth16<E, C> {
```

Comment:
As the groth16 and plonky are seperated into two crates, seems this trait should be remove.